kubemarine/procedures/check_iaas.py

```diff
@@ -276,6 +277,41 @@ def system_distributive(cluster):
         tc.success(results=", ".join(detected_supported_os))
 
 
+def check_access_to_thirdparties(cluster: KubernetesCluster):
+    detect_preinstalled_python(cluster)
+    broken = []
+
+    # Load script for checking sources
+    all_group = cluster.nodes['all']
+    local_path = utils.get_resource_absolute_path("resources/scripts/check_thirdparty_avaliability.py",
+                                                  script_relative=True)
+    random_temp_path = "/tmp/%s.py" % uuid.uuid4().hex
+    all_group.put(local_path, random_temp_path, sudo=True, binary=False)
+
+    for destination, config in cluster.inventory['services'].get('thirdparties', {}).items():
+        # Check if curl
+        if config['source'][:4] != 'http' or '://' not in config['source'][4:8]:
+            continue
+        # Check with script
+        common_group = cluster.create_group_from_groups_nodes_names(config.get('groups', []), config.get('nodes', []))
+        for node in common_group.get_ordered_members_list(provide_node_configs=True):
+            python_executable = cluster.context['nodes'][node['connect_to']]['python']['executable']
+            res = node['connection'].sudo("%s %s %s %s" % (python_executable, random_temp_path, config['source'],
```

Comment:
Looks like we could avoid use run instead of sudo. The same is in rm instead of sudo rm below

Reply:
Sorry, I was initially misleading. It is related not only to NodeGroup.sudo, but also to all_group.sudo(rm) below and to all_group.put(sudo=True) above.